### PR TITLE
Runner test refactor

### PR DESCRIPTION
**Pull request by macfarla**
Refactor to make it easier to find the cause when this tests fails. 
* Deduplicated code. 
* Added a call to peerCount to expose peering issues vs syncing issues
* Because of the async wait() you don't get the exact line where the assertion fails. Added comments for common spots where it fails to make this easier. 
* Also 2 minutes is plenty. 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: besu/src/test/java/org/hyperledger/besu/RunnerTest.java [hunk 1]
```diff
@@ -393,6 +345,24 @@ private void syncFromGenesis(final SyncMode mode, final GenesisConfigFile genesi
...
   }
 
+  private int getNumber(final Response response) throws IOException {
+    final int currentBlock =
+        UInt256.fromHexString(new JsonObject(response.body().string()).getString("result"))
+            .intValue();
+    return currentBlock;
+  }
+
+  @NotNull
```
Comment (garyschulte) [inline besu/src/test/java/org/hyperledger/besu/RunnerTest.java:355]: what is @NotNull accomplishing in this case?  it is a jetbrains annotation, and won't be checked by findbugs or CI AFAIK.  Also this is on a test method that returns the result of a builder.  

I ask everytime I see the jetbrains import.

File: besu/src/test/java/org/hyperledger/besu/RunnerTest.java [hunk 2]
```diff
@@ -301,61 +261,53 @@ private void syncFromGenesis(final SyncMode mode, final GenesisConfigFile genesi
...
-                                        "{\"jsonrpc\":\"2.0\",\"id\":"
-                                            + Json.encode(7)
-                                            + ",\"method\":\"eth_blockNumber\"}"))
-                                .url(baseUrl)
-                                .build())
-                        .execute()) {
-
-                  assertThat(resp.code()).isEqualTo(200);
+                try (final Response blockNumberResp =
+                    client.newCall(getRequest("eth_blockNumber", baseUrl)).execute()) {
```
Comment (garyschulte) [inline besu/src/test/java/org/hyperledger/besu/RunnerTest.java:269]: :raised_hands:

**Review by garyschulte — APPROVED** [review]
nice refactor.  just a question about the purpose of the annotation

File: besu/src/test/java/org/hyperledger/besu/RunnerTest.java (hunk 1 above)
Comment (macfarla) [reply besu/src/test/java/org/hyperledger/besu/RunnerTest.java:355]: 👍 good point. removed

**Pull request merged**